source/nanoFramework.Hardware.Esp32/TouchPad/TouchPadBase.cs

```diff
@@ -0,0 +1,154 @@
+﻿using System;
+using System.Collections;
+using System.Runtime.CompilerServices;
+
+namespace nanoFramework.Hardware.Esp32.TouchPad
+{
+  /// <summary>
+  /// Base class for toch pad sensors: interrupt or polling driven, using single or multiple gpio pins
+  /// </summary>
+  public abstract class TouchPadBase // : IDisposable
+  {
+    //code review: Gpio​Pin uses it, but it's internal to Windows.Devices.Gpio  - please review if it applicable here
```

Comment:
Agreed: tt makes sense to have a specific event for touchpads. This has to be added in the events class lib first.